packages/salesforcedx-vscode-core/src/commands/forceApexExecute.ts

```diff
@@ -4,99 +4,96 @@
-} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import { ExecuteService } from '@salesforce/apex-node';
+import { Connection } from '@salesforce/core';
 import {
   CancelResponse,
   ContinueResponse,
   ParametersGatherer
 } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
-import * as fs from 'fs';
-import * as path from 'path';
 import * as vscode from 'vscode';
+import { channelService } from '../channels';
 import { nls } from '../messages';
-import { getRootWorkspacePath, hasRootWorkspace } from '../util';
+import { notificationService } from '../notifications';
+import { telemetryService } from '../telemetry';
+import { hasRootWorkspace } from '../util';
 import {
+  ApexLibraryExecutor,
   SfdxCommandlet,
-  SfdxCommandletExecutor,
   SfdxWorkspaceChecker
 } from './util';
 
-class ForceApexExecuteExecutor extends SfdxCommandletExecutor<{}> {
-  public build(data: TempFile): Command {
-    return new SfdxCommandBuilder()
-      .withDescription(nls.localize('force_apex_execute_document_text'))
-      .withArg('force:apex:execute')
-      .withFlag('--apexcodefile', data.fileName)
-      .withLogName('force_apex_execute')
-      .build();
-  }
-}
-
-class CreateApexTempFile implements ParametersGatherer<{ fileName: string }> {
+export class AnonApexGatherer
+  implements ParametersGatherer<{ fileName?: string; apexCode?: string }> {
   public async gather(): Promise<
-    CancelResponse | ContinueResponse<{ fileName: string }>
+    CancelResponse | ContinueResponse<{ fileName?: string; apexCode?: string }>
   > {
     if (hasRootWorkspace()) {
-      const fileName = path.join(
-        getRootWorkspacePath(),
-        '.sfdx',
-        'tools',
-        'tempApex.input'
-      );
-      const editor = await vscode.window.activeTextEditor;
-
+      const editor = vscode.window.activeTextEditor;
       if (!editor) {
         return { type: 'CANCEL' };
       }
 
-      let writeFile;
       const document = editor.document;
-
-      if (editor.selection.isEmpty) {
-        writeFile = await writeFileAsync(fileName, document.getText());
-      } else {
-        writeFile = await writeFileAsync(
-          fileName,
-          document.getText(editor.selection)
-        );
+      if (!editor.selection.isEmpty) {
+        return {
+          type: 'CONTINUE',
+          data: { apexCode: document.getText(editor.selection) }
+        };
       }
 
-      return writeFile
-        ? { type: 'CONTINUE', data: { fileName } }
-        : { type: 'CANCEL' };
+      return { type: 'CONTINUE', data: { fileName: document.uri.fsPath } };
     }
     return { type: 'CANCEL' };
   }
 }
 
-type TempFile = {
-  fileName: string;
-};
+const workspaceChecker = new SfdxWorkspaceChecker();
+const fileNameGatherer = new AnonApexGatherer();
+
+export class ApexLibraryExecuteExecutor extends ApexLibraryExecutor {
+  protected executeService: ExecuteService | undefined;
+
+  public createService(conn: Connection): void {
```

Comment:
This part highlights a current limitation with the library in that we need to individually initialize every service in the library before using it. We have flows in which we'll need to run more than one service from the library sequentially (e.g. the new Replay Debugger) and we should only have to initialize (pass a connection object) once and be able to use as many services as needed.

We don't need to solve it for this PR but we should have a work item to track those changes.

Reply:
Agreed, we should have a single entry point for the consumer to access any methods in the library. Here's the work item to track that: https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008HWFgIAO/view